connectors/spark/src/main/scala/io/unitycatalog/spark/auth/GenericCredentialProvider.java

```diff
@@ -123,31 +132,82 @@ private GenericCredential renewCredential() throws ApiException {
     }
   }
 
+  private boolean isRecoverable(Throwable e) {
+    if (e instanceof ApiException) {
```

Comment:
Where can I find those recoverable codes ?  And how could we confidently say that we've already covered all the recoverable code ?

Reply:
Please also consider how to design test to cover those recoverable codes, thanks .